Sources/Controllers/2.0/Settings/OAImportDuplicatesViewController.mm

```diff
@@ -0,0 +1,518 @@
+#define kTitleTwoIconsRoundCell @"OATitleTwoIconsRoundCell"
+#define kCellTypeWithActivity @"OAActivityViewWithTitleCell"
+#define RENDERERS_DIR @"rendering/"
+#define ROUTING_PROFILES_DIR @"routing/"
+
+
+@interface HeaderType : NSObject
+@property (nonatomic) NSString *label;
+@end
+
+@implementation HeaderType
+- (instancetype) initWithLabel:(NSString *)label
+{
+    self = [super init];
+    if (self)
+    {
+        _label = label;
+    }
+    return self;
+}
+@end
+
+
+@interface OAImportDuplicatesViewController () <UITableViewDelegate, UITableViewDataSource, OASettingsImportExportDelegate>
+
+@end
+
+@implementation OAImportDuplicatesViewController
+{
+    OsmAndAppInstance _app;
+    NSArray<id> *_duplicatesList;
+    NSArray<OASettingsItem *> * _settingsItems;
+    NSString *_file;
+    OASettingsHelper *_settingsHelper;
+    
+    NSString *_title;
+    NSString *_description;
+
+    NSMutableArray<NSMutableArray<NSDictionary *> *> *_data;
+}
+
+- (instancetype) init
+{
+    self = [super init];
+    if (self)
+    {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (instancetype) initWithDuplicatesList:(NSArray<id> *)duplicatesList settingsItems:(NSArray<OASettingsItem *> *)settingsItems file:(NSString *)file
+{
+    self = [super init];
+    if (self)
+    {
+        _duplicatesList = duplicatesList;
+        _settingsItems = settingsItems;
+        _file = file;
+        [self commonInit];
+    }
+    return self;
+}
+
+//onCreate
+- (void) commonInit
+{
+    _app = [OsmAndApp instance];
+    _settingsHelper = [OASettingsHelper sharedInstance];
+    
+    OAImportAsyncTask *importTask = _settingsHelper.importTask;
+    if (!importTask)
+        _settingsItems = [importTask getSelectedItems];
+    if (!_duplicatesList)
+        _duplicatesList = [importTask getDuplicates];
+    if (!_file)
+        _file = [importTask getFile];
+    
+    importTask.delegate = self;
+}
+
+//onActivityCreated
+- (void) viewDidLoad
+{
+    self.tableView.delegate = self;
+    self.tableView.dataSource = self;
+    self.tableView.allowsSelection = NO;
+    self.additionalNavBarButton.hidden = YES;
+    [self setupBottomViewMultyLabelButtons];
+    [super viewDidLoad];
+    
+    if (_duplicatesList)
+        [self prepareData: [self prepareDuplicates:_duplicatesList]];
+    if ([_settingsHelper.importTask getImportType] == EOAImportTypeImport)
+        [self setupImportingUI];
+    else
+        _title = OALocalizedString(@"import_duplicates_title");
+    _title = OALocalizedString(@"import_duplicates_title");
+}
+
+- (void) setupImportingUI
+{
+    _title = OALocalizedString(@"shared_string_importing");
+    _description = [NSString stringWithFormat:OALocalizedString(@"importing_from"), _file];
+    
+    [self turnOnLoadingIndicator];
+    self.bottomBarView.hidden = YES;
+    self.view.backgroundColor = self.tableView.backgroundColor;
+    [self.tableView reloadData];
+    
+}
+
+- (void) turnOnLoadingIndicator
+{
+    _data = @[ @[ @{
```

Comment:
assigning immutable array to NSMutableArray<NSMutableArray<NSDictionary *> *> *_data